libs/sdk-react-native/android/src/main/java/com/breezsdk/BreezSDKModule.kt

```diff
@@ -146,6 +164,46 @@ class BreezSDKModule(reactContext: ReactApplicationContext) : ReactContextBaseJa
         }
     }
 
+    @ReactMethod
+    fun signMessage(reqData: ReadableMap, message: String, promise: Promise) {
+        executor.execute {
+            val signMessageRequest = asSignMessageRequest(reqData)
+            if signMessageRequest == null {
+                 promise.reject(TAG, "Invalid reqData")
+            } else {
+                try {
+                    val response = getBreezServices().sign_message(signMessageRequest)
+                    val signedMessage = response.signed_message
+                    promise.resolve(signedMessage)
+                } catch (e: SdkException) {
+                    e.printStackTrace()
+                    promise.reject(TAG, e.message ?: "Error calling signMessage", e)
+                }
+            }
+        }
+    }
+
+    @ReactMethod
+    fun checkMessage(reqData: ReadableMap, message: String, signature: String, pubkey: String, promise: Promise) {
+        executor.execute {
+            val checkMessageRequest = asCheckMessageRequest(reqData)
+            if checkMessageRequest == null {
+                 promise.reject(TAG, "Invalid reqData")
+            } else {
+                try {
+                    val response = getBreezServices().check_message(checkMessageRequest)
+                    val signedMessage = response.signed_message
+                    promise.resolve(signedMessage)
```

Comment:
The CheckMessageResponse object should be returned here, you can add a mapping function to `BreezSDKMapper.kt`

```
fun readableMapOf(checkMessageResponse: CheckMessageResponse): ReadableMap {
    return readableMapOf(
            "isValid" to checkMessageResponse.isValid
    )
}
```